src/Orleans/LogConsistency/ILogConsistencyProvider.cs

```diff
@@ -30,14 +30,20 @@ public interface ILogViewAdaptorFactory
         /// <summary> Returns true if a storage provider is required for constructing adaptors. </summary>
         bool UsesStorageProvider { get; }
 
+
+        /// <summary> Returns true if a event-storage provider is required for constructing adaptors. </summary>
+        bool UsesEventStorageProvider { get; }
```

Comment:
Absolutely, as I mentioned [here](https://github.com/dotnet/orleans/pull/2711#pullrequestreview-43040619) a lot of this can be simplified once we add the GrainActivationContext from https://github.com/dotnet/orleans/pull/2856
Not sure there's much he can do in this PR until that's in though